### PR TITLE
added minification for esbuild

### PR DESCRIPTION
**Pull request by lucamezzalira**
*Issue #, if available:*

*Description of changes:*
I've set up code minification and platform optimization for esbuild so the code size per Lambda moved from 1.5mb to 700kb
I've also bump up the CDK version to use the latest node runtime (node.js 16) that wasn't available in CDK v2.0.0
Finallly I collected the common Lambda functions configurations into variables so we can apply changes to all the functions without copying and pasting a change

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/serverless-typescript-demo-stack.ts [hunk 1]
```diff
@@ -33,20 +24,40 @@ export class ServerlessTypescriptDemoStack extends Stack {
...
+      POWERTOOLS_SERVICE_NAME: 'serverless-typescript-demo',
+      POWERTOOLS_LOGGER_LOG_LEVEL: 'WARN',
+      POWERTOOLS_LOGGER_SAMPLE_RATE: '0.01',
+      POWERTOOLS_LOGGER_LOG_EVENT: 'true',
+      POWERTOOLS_METRICS_NAMESPACE: 'AwsSamples',
+  };
+
+    const esBuildSettings = {
+      minify: true,
+      platform: "node"
```
Comment (everett1992) [inline lib/serverless-typescript-demo-stack.ts]: [platform=node](https://github.com/aws/aws-cdk/blob/v1.165.0/packages/%40aws-cdk/aws-lambda-nodejs/lib/bundling.ts#L183-L184) is the default and I don't think BundlerSettings#platform is read, it's not a [documented option.](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-lambda-nodejs.BundlingOptions.html)

Comment (everett1992) [reply lib/serverless-typescript-demo-stack.ts]: I recommend removing `"strictPropertyInitialization": false` and the other `strict` overrides from tsconfig. 
Then add `BundlingOptions` type annotation to L36 and you should get a type error on platform. 

https://github.com/aws-samples/serverless-typescript-demo/blob/main/tsconfig.json#L21

**Comment by dreamorosi** [comment]
May I suggest to also add `esbuild` to the dependencies of the project so that [the bundling is done locally](https://docs.aws.amazon.com/cdk/api/v1/docs/aws-lambda-nodejs-readme.html#local-bundling) on the host machine rather than involving Docker? I was going to submit a PR in the next days but since you're already changing this it could be a good time.

**Comment by lucamezzalira** [comment]
added, thanks for the contribution @dreamorosi

**Pull request merged**